ibu-imager/seedcleanup/seedcleanup.go

```diff
@@ -0,0 +1,174 @@
+	ops                  ops.Ops
+	backupDir            string
+	containerRegistry    string
+	authFile             string
+	recertContainerImage string
+	recertSkipValidation bool
+}
+
+func NewSeedCleaner(log *logrus.Logger, ops ops.Ops, backupDir,
+	containerRegistry, authFile, recertContainerImage string, recertSkipValidation bool) *SeedCleaner {
+
+	return &SeedCleaner{
+		log:                  log,
+		ops:                  ops,
+		backupDir:            backupDir,
+		containerRegistry:    containerRegistry,
+		authFile:             authFile,
+		recertContainerImage: recertContainerImage,
+		recertSkipValidation: recertSkipValidation,
+	}
+}
+
+// CleanupSeedCluster comprises the Imager workflow for cleanup operations after creating a seed image
+// out of an SNO cluster.
+func (s *SeedCleaner) CleanupSeedCluster() error {
+	s.log.Println("Cleaning up seed cluster")
+
+	if err := s.cleanupBackupDir(); err != nil {
+		return err
+	}
+
+	if err := s.cleanupConfigurationFiles(); err != nil {
+		return err
+	}
+
+	if s.recertSkipValidation {
+		s.log.Info("Skipping restoring certificates in etcd via recert tool.")
+	} else {
+		if err := s.restoreCertificates(); err != nil {
+			return err
+		}
+	}
+
+	if err := s.restoreCluster(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *SeedCleaner) cleanupBackupDir() error {
+	if err := os.RemoveAll(s.backupDir); err != nil {
+		return fmt.Errorf("error removing %s: %w", s.backupDir, err)
+	}
+
+	if err := os.Remove(common.BackupChecksDir + "/container_list.done"); err != nil && !os.IsNotExist(err) {
+		s.log.Errorf("error removing %s: %v", common.BackupChecksDir+"/container_list.done", err)
+	}
+
+	s.log.Info("Backup data removed successfully.")
+	return nil
+}
+
+func (s *SeedCleaner) cleanupConfigurationFiles() error {
+	for _, systemdUnit := range common.SystemdUnitsToDisable {
+		if _, err := s.ops.SystemctlAction("disable", systemdUnit); err != nil {
+			s.log.Errorf("error disabling %s unit: %v", systemdUnit, err)
+		}
+	}
+	s.log.Info("Preparation systemd units disabled successfully.")
+
+	for _, file := range common.FilesToRemove {
+		if err := os.Remove(file); err != nil && !os.IsNotExist(err) {
+			s.log.Errorf("error removing %s: %v", file, err)
+		}
+	}
+	s.log.Info("Preparation systemd scripts removed successfully.")
+
+	return nil
+}
+
+func (s *SeedCleaner) restoreCertificates() error {
+	etcdImage, err := s.ops.GetImageFromPodDefinition(common.EtcdStaticPodFile, common.EtcdStaticPodContainer)
+	if err != nil {
+		return fmt.Errorf("error getting available etcd image: %w", err)
+	}
+
+	if err := s.ops.RunUnauthenticatedEtcdServer(etcdImage, s.authFile); err != nil {
+		return err
+	}
+
+	defer func() {
+		if _, err = s.ops.RunInHostNamespace(
+			"podman", []string{"kill", "recert_etcd"}...); err != nil {
+			s.log.Errorf("Failed to kill recert_etcd container: %v", err)
+		}
+		s.log.Info("Unauthenticated etcd server and recert containers removed successfully.")
+	}()
```

Comment:
We should stuff all this "get image, run etcd, defer stop etcd, run recert" logic in a package, as it's repeated  in the seedcreator as well

Reply:
Agree, will move these steps to the [recert](https://github.com/openshift-kni/lifecycle-agent/blob/main/internal/recert/recert.go) package.